wizard/state.py

```diff
@@ -53,3 +62,75 @@ class Metadata:
 
     def __bool__(self):
         return self.name != "" and self.url != ""
+    
+@dataclasses.dataclass
+class Croissant:
+    metadata: Metadata = Metadata()
+    distributions: list[Distribution] = dataclasses.field(default_factory=list)
+    record_sets: list[RecordSet] = dataclasses.field(default_factory=list)
+    def update_metadata(self, metadata: Metadata) -> None:
+        self.metadata = Metadata
+    def add_distribution(self, distribution: Distribution) -> None:
+        self.distributions.append(distribution)
+    def update_distribution(self, key: int, distribution: Distribution) -> None:
+        self.distributions[key] = distribution
+    def remove_distribution(self, key: int) -> None:
+        del self.distributions[key]
+    def add_record_set(self, record_set: RecordSet) -> None:
+        self.record_sets.append(record_set)
+    def update_record_set(self, key: int, record_set: RecordSet) -> None:
+        self.record_sets[key] = record_set
+    def remove_record_set(self, key: int) -> None:
+        del self.record_sets[key]
+
+def CanonicalToWizard(dataset: mlc.Dataset) -> Croissant:
+    canonical_metadata = dataset.metadata
+    metadata = Metadata(
+        name=canonical_metadata.name,
+        description=canonical_metadata.description,
+        citation=canonical_metadata.citation,
+        license=canonical_metadata.license,
+        url=canonical_metadata.url
+    )
+    distributions = []
+    for file in canonical_metadata.distribution:
+        if isinstance(file, mlc.nodes.FileObject):
+            distributions.append(Distribution(
```

Comment:
Here we see that it's interesting to have a distinction between FileObject and FileSets. If you keep the names `FileObject`/`FileSet` and copy mlcroissant, you reduce the mental overload and you allow easy checks:

```
if isinstance(file, FileObject):

# or

if isinstance(file, FileSet):
```

Probably FileObject/FileSet will be handled by different Streamlit components so this distinction would be interesting.

Reply:
added both, however, for the moment, it seems like everything addable from the application is a FileObject, but this was the conversation we were having this morning I believe.

Reply:
Correct!